app/src/test/java/com/codesoom/assignment/application/ProductServiceTest.java

```diff
@@ -0,0 +1,129 @@
+        productService = new ProductService(productRepository);
+        product = new Product(NAME, MAKER, PRICE, IMAGE_URL);
+    }
+
+    @Nested
+    @DisplayName("createProduct 메소드는")
+    class Describe_createProduct {
+        ProductCommandDto productCommandDto;
+
+        @BeforeEach
+        void setUp() {
+            productCommandDto = ProductCommandDto.builder()
+                    .name(NAME)
+                    .maker(MAKER)
+                    .price(PRICE)
+                    .imageUrl(IMAGE_URL).build();
+        }
+
+        @Test
+        @DisplayName("생성된 product의 dto를 반환한다.")
+        void it_returns_created_product() {
+            ProductDto newProduct = productService.createTask(productCommandDto);
+
+            assertThat(newProduct.getId()).isNotNull();
+            assertThat(newProduct.getName()).isEqualTo(NAME);
+            assertThat(newProduct.getMaker()).isEqualTo(MAKER);
+            assertThat(newProduct.getPrice()).isEqualTo(PRICE);
+            assertThat(newProduct.getImageUrl()).isEqualTo(IMAGE_URL);
+        }
+    }
+
+    @Nested
+    @DisplayName("updateProduct 메소드는")
+    class Describe_updateProduct {
+        private static final String UPDATE_NAME = "캣닢";
+        private static final String UPDATE_MAKER = "캣드러그";
+        private static final int UPDATE_PRICE = 7000;
+        private static final String UPDATE_IMAGE_URL = "https://www.catdrug.com/images/1234567";
+
+        private Long id;
+        private ProductCommandDto productCommandDto;
+
+        @BeforeEach
+        void setUp() {
+            productCommandDto = ProductCommandDto.builder()
+                    .name(UPDATE_NAME)
+                    .maker(UPDATE_MAKER)
+                    .price(UPDATE_PRICE)
+                    .imageUrl(UPDATE_IMAGE_URL)
+                    .build();
+        }
+
+        @DataJpaTest
+        @Nested
+        @DisplayName("존재하는 id가 주어지면")
+        class Context_with_existed_id {
+
+            @BeforeEach
+            void setUp() {
+                Product newProduct = productRepository.save(product);
+                id = newProduct.getId();
+            }
+
+            @Test
+            @DisplayName("product를 변경한다.")
+            void it_updates_product() {
+                productService.updateProduct(id, productCommandDto);
+
+                Product updatedProduct = productRepository.findById(id).get();
+
+                assertThat(updatedProduct.getName()).isEqualTo(UPDATE_NAME);
+                assertThat(updatedProduct.getMaker()).isEqualTo(UPDATE_MAKER);
+                assertThat(updatedProduct.getPrice()).isEqualTo(UPDATE_PRICE);
+                assertThat(updatedProduct.getImageUrl()).isEqualTo(UPDATE_IMAGE_URL);
+            }
+        }
+
+        @DataJpaTest
+        @Nested
+        @DisplayName("존재하지 않는 id가 주어지면")
+        class Context_with_non_existed_id {
+
+            @BeforeEach
+            void setUp() {
+                id = 100L;
+            }
+
```

Comment:
```suggestion
             private final Long 존재하지_않는_ID = 100L;
```

더 넓은 스코프에 id를 넣어서 계속 바꾸면서 사용하기보다는 스코프를 줄이고 그에 맞는 변수 이름을 사용하는게 복잡성을 줄이는데 도움이 되요!